src/Microsoft.AspNetCore.OData/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.AspNet.OData.Interfaces;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+        internal const string ODataServiceVersionHeader = "OData-Version";
+        internal const string ODataMaxServiceVersionHeader = "OData-MaxVersion";
+        internal const ODataVersion DefaultODataVersion = ODataVersion.V4;
+
+        /// <summary>
+        /// Gets the <see cref="IODataFeature"/> from the services container.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <returns>The <see cref="IODataFeature"/> from the services container.</returns>
+        public static IODataFeature ODataFeature(this HttpRequest request)
+        {
+            if (request == null)
+            {
+                throw Error.ArgumentNull("request");
+            }
+
+            return request.HttpContext.ODataFeature();
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IETagHandler"/> from the services container.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <returns>The <see cref="IETagHandler"/> from the services container.</returns>
+        public static IETagHandler ETagHandler(this HttpRequest request)
+        {
+            if (request == null)
+            {
+                throw Error.ArgumentNull("request");
+            }
+
+            return request.HttpContext.ETagHandler();
+        }
+
+        /// <summary>
+        /// Gets the <see cref="IODataPathHandler"/> from the request container.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <returns>The <see cref="IODataPathHandler"/> from the request container.</returns>
+        public static IODataPathHandler GetPathHandler(this HttpRequest request)
```

Comment:
maybe we use same naming convention between 'ETaghandler" and "GetPathHandler"?

Reply:
I opted for the Get...() pattern except for ODataFeature() because of ODataProperties() on the AspNet side.